src/wxProfilerGUI/profilergui.h

```diff
@@ -66,24 +66,79 @@ struct AttachInfo
 
 wxBitmap LoadPngResource(const wchar_t *szName, const wxWindowBase* w);
 
+// Encapsulate a (config) value that can be overridden.
+//   `T` must be copyable and have a (sensible) default constructor.
+template<typename T>
+class OverridableOption
+{
+public:
+	OverridableOption()
+		: is_overridden(false)
+	{}
+	
+	OverridableOption( const T& config_value )
+		: config_value(config_value), is_overridden(false)
+	{}
+
+	// setters
+	void SetConfigValue( const T& new_value )
+	{
+		is_overridden = false;

```

Comment:
I am on the fence if this is a good idea.

On one hand, if the profiler behaves some way, then you go into the settings, *don't change anything*, save & exit, and then the profiler behaves some other way, that's bad.

On the other hand, if you go into the settings and clearly see that the settings don't correspond to the profiler's current behavior, that's also bad.

Ideally the settings should indicate that some values are overridden. Maybe a pop-up message box shown when they're opened is sufficient.

Anyway, this is a pretty minor thing to worry about.

Reply:
In order for me to observe what you're describing, I'd have to specify command-line parameters when I invoke the app, and then go into the settings. This seems like a very niche case to me.

Is there a truly _correct_ way to handle this?

Perhaps I could disable the GUI controls in the settings if the item is specified on the command line (as well as ignore the value of the control when we update the settings when the window closes).